src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java

```diff
@@ -145,9 +147,10 @@ public TenantSecurityClient(String tspDomain, String apiKey, int requestThreadSi
     }
 
     this.encryptionExecutor = Executors.newFixedThreadPool(aesThreadSize);
-
     this.encryptionService =
         new TenantSecurityRequest(tspDomain, apiKey, requestThreadSize, timeout);
+    this.deterministicClient = new DeterministicTenantSecurityClient(tspDomain, apiKey,
```

Comment:
When we create it this way, we don't give them the ability to create 2 different sized pools for crypto operations. It seems like in this case it should actually pass the shared pools to the deterministic client.

Reply:
I made a new protected `DeterministicTenantSecurityClient` constructor that just takes the AES `ExecutorService` and the `TenantSecurityRequest` from the existing client so it can share pools. I then added a comment to the `getDeterministicClient()` function saying that it shares pools, but you can access the constructor directly if that's not what you want.